vnext/Microsoft.ReactNative/IJsiNonAbiHostObject.idl

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "DocString.h"
+
+namespace Microsoft.ReactNative
+{
+  [webhosthidden]
+  DOC_STRING(
+    "The `IJsiNonAbiHostObject` helps with passing non-ABI C++ TurboModules "
+    "via @ReactModuleProvider. Since React Native Windows already uses @IReactNonAbiValue "
+    "internally to represent attributed native modules, this wrapper is needed to have an "
+    "alternate interface to type check against.")
+  interface IJsiNonAbiHostObject {
+    DOC_STRING("Gets a value indicating whether the object has a value. This is just a dummy method to avoid issues with empty interfaces in C++/WinRT.")
+    Boolean HasValue { get; };
```

Comment:
@acoates-ms do you have any pointers to where I should add a GUID to avoid this dummy property?

Also, do you have any suggestions for how we can shove a C++ TurboModule into an IInspectable (to match the IReactModuleBuilder delegate type) and be able to query *some* interface to decide that it's actually a JsiNonAbiHostObject? If so, we won't need this IDL. Bear in mind that IReactNonAbiValue is already "taken" for NM2 modules.

Reply:
How to add a guid, to remove the dummy property:
https://github.com/microsoft/react-native-windows/blob/20a4c9b10681bef32c245066c1bb170e0d60dd5e/vnext/Microsoft.ReactNative/CompositionSwitcher.idl#L25C8-L25C8